src/network/networkd-sysctl.c

```diff
@@ -40,10 +42,69 @@ static void manager_set_ip_forwarding(Manager *manager, int family) {
                                   enable_disable(t), af_to_ipv4_ipv6(family));
 }
 
+static int sysctl_inotify(sd_event_source *source, const struct inotify_event *event, void *userdata) {
+        _cleanup_free_ char *buf = NULL, *path = NULL;
+        Hashmap *sysctl_shadow = ASSERT_PTR(userdata);
+        const char *dir_path, *value;
+        int r;
+
+        assert(event);
+
+        r = sd_event_source_get_inotify_path(source, &dir_path);
+        if (r < 0)
+                return log_debug_errno(r, "Failed to get the watched inotify path: %m");
+
+        path = path_join(dir_path, event->name);
+        if (!path)
+                return log_oom_debug();
+
+        value = hashmap_get(sysctl_shadow, path);
+
+        /* We don't handle this sysctl */
+        if (!value)
+                return r;
+
+        r = read_virtual_file(path, SIZE_MAX, &buf, NULL);
+        if (r < 0)
+                return log_debug_errno(r, "Failed to read sysctl value '%s': %m", path);
+
+        delete_trailing_chars(buf, "\n");
+
+        if (!streq(value, buf))
+                log_warning("sysctl %s was changed from '%s' to '%s'", path, value, buf);
+
+        return 0;
+}
+
+static int manager_inotify_add(Manager *manager) {
+        sd_event_source **source;
+        int r;
+
+        assert(manager);
+
+        source = manager->sysctl_event_sources;
+
+        FOREACH_STRING(p,
+                "/proc/sys/net/ipv4/conf/all",
+                "/proc/sys/net/ipv4/conf/default",
+                "/proc/sys/net/ipv6/conf/all",
+                "/proc/sys/net/ipv6/conf/default") {
```

Comment:
Maybe put these in an array and then add a compile time assertion with `assert_cc()`  that the size of this array is the same as the size of the event sources array in the `Manager` struct? That ensures that both arrays always have the same size.

Reply:
Ok, but the I will need to iterate them with `FOREACH_ELEMENT()` instead, as `FOREACH_STRING()` seems to only accept an inline string list